test/integration/base_integration_test.cc

```diff
@@ -335,7 +335,7 @@ void BaseIntegrationTest::createGeneratedApiTestServer(
       bootstrap_path, version_, on_server_ready_function_, on_server_init_function_,
       deterministic_value_, timeSystem(), *api_, defer_listener_finalization_, process_object_,
       validator_config, concurrency_, drain_time_, drain_strategy_, proxy_buffer_factory_,
-      use_real_stats_);
+      use_real_stats_, skip_asserts_);
```

Comment:
rather than passing this bool around can you just call `SkipAsserts::skip()` where you need to?

I realize I am normally anti-singleton like this but maybe that one is different than the ones I usually complain about in that it's really just advanced clang-tidy/lint in active code form, so it's kind of meta (sic) to the functionality of the proxy.

Reply:
skip() just tells you if you _should_ skip.  it returns true if that thread has a SkipAssert on stack (since that increments a variable on creation, and decrements it ondeletion).  So yes, I need actual SkipAsserts in the test threads, and because it needs to be instantiated in the test thread, I need to plumb the booleans.

Reply:
Oh sorry -- I designed that and completely forgot about it :)

Reply:
the number of times I've thought "who wrote _this_ confusing bit of code?  Oh wait, that was me" :-P